libkbfs/file_data.go

```diff
@@ -509,44 +535,298 @@ func (fd *fileData) createIndirectBlock(
 	return fblock, nil
 }
 
-// newRightBlock appends a new block (with a temporary ID) to the end
-// of this file.
+// newRightBlock creates space for a new rightmost block, creating
```

Comment:
So this function doesn't append to the end of the file anymore, right? I guess the name can still remain the same, (I can't think of a better one) but I think the comment can be a bit clearer about that. Perhaps "creates space for a new block to the right of (but not necessarily adjacent to) the leaf block pointed to by `parentBlocks`..."